app/services/schools/advice/base_service.rb

```diff
@@ -0,0 +1,26 @@
+module Schools
+  module Advice
+    class BaseService
+      def initialize(school, meter_collection)
+        @school = school
+        @meter_collection = meter_collection
+      end
+
+      def date_range(meter: aggregate_meter)
+        [meter.amr_data.start_date, meter.amr_data.end_date]
+      end
+
+      # from analytics: lib/dashboard/charting_and_reports/content_base.rb
+      def calculate_rating_from_range(good_value, bad_value, actual_value)
+        actual_value = actual_value.abs
+        [10.0 * [(actual_value - bad_value) / (good_value - bad_value), 0.0].max, 10.0].min.round(1)
```

Comment:
I realise this is bought over from the analytics but I'm thinking this method would benefit from a comment on what it's returning and why (and the use the abs of `actual_value`).  or maybe it could go in as a `describe` block in the spec.

Reply:
I'll add a comment. I'm also thinking that the ratings may need to be pushed down into the analytics services.